src/datatable/datatable.jl

```diff
@@ -790,26 +790,24 @@ end
 ##
 ##############################################################################
 
-function categorical!(dt::DataTable, cname::@compat(Union{Integer, Symbol}), compact::Bool=true)
+function categorical!(dt::DataTable, cname::Union{Integer, Symbol}, compact::Bool=true)
     dt[cname] = categorical(dt[cname], compact)
-    return
+    dt
 end
 
-function categorical!{T <: @compat(Union{Integer, Symbol})}(dt::DataTable, cnames::Vector{T},
-                                                            compact::Bool=true)
+function categorical!{T <: Union{Integer, Symbol}}(dt::DataTable, cnames::Vector{T},
+                                                   compact::Bool=true)
     for cname in cnames
         dt[cname] = categorical(dt[cname], compact)
     end
-    return
+    dt
 end
 
 function categorical!(dt::DataTable, compact::Bool=true)
     for i in 1:size(dt, 2)
-        if eltype(dt[i]) <: AbstractString
```

Comment:
Please don't change this in this PR, it's completely unrelated and it affects user-visible behavior. I don't think that function would be very useful if it converted numeric strings to `Categorical`...